lib/Optimizer/Transforms/GetConcreteMatrix.cpp

```diff
@@ -0,0 +1,95 @@
+/*******************************************************************************
+ * Copyright (c) 2022 - 2024 NVIDIA Corporation & Affiliates.                  *
+ * All rights reserved.                                                        *
+ *                                                                             *
+ * This source code and the accompanying materials are made available under    *
+ * the terms of the Apache License 2.0 which accompanies this distribution.    *
+ ******************************************************************************/
+
+#include "PassDetails.h"
+#include "cudaq/Optimizer/Builder/Intrinsics.h"
+#include "cudaq/Optimizer/CodeGen/Passes.h"
+#include "cudaq/Optimizer/Dialect/CC/CCOps.h"
+#include "cudaq/Optimizer/Dialect/Quake/QuakeOps.h"
+#include "cudaq/Optimizer/Transforms/Passes.h"
+#include "mlir/Dialect/LLVMIR/LLVMDialect.h"
+#include "mlir/Dialect/LLVMIR/LLVMTypes.h"
+#include "mlir/Transforms/DialectConversion.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+#include "mlir/Transforms/Passes.h"
+
+namespace cudaq::opt {
+#define GEN_PASS_DEF_GETCONCRETEMATRIX
+#include "cudaq/Optimizer/Transforms/Passes.h.inc"
+} // namespace cudaq::opt
+
+#define DEBUG_TYPE "get-concrete-matrix"
+
+using namespace mlir;
+
+namespace {
+
+class CustomUnitaryPattern
+    : public OpRewritePattern<quake::CustomUnitarySymbolOp> {
+
+public:
+  using OpRewritePattern::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(quake::CustomUnitarySymbolOp customOp,
+                                PatternRewriter &rewriter) const override {
+    auto generator = customOp.getGenerator();
+
+    auto parentModule = customOp->getParentOfType<ModuleOp>();
+    auto funcOp = parentModule.lookupSymbol<func::FuncOp>(generator);
+    if (!funcOp) {
+      return failure();
+    }
+
+    StringRef concreteMatrix;
+    mlir::Region &body = funcOp.getBody();
```

Comment:
Shouldn't this just be a `funcOp.walk([&](AddressOfOp op) {...})`

Reply:
I didn't know this was possible, I will update. Thanks!

Reply:
Addressed in 85acc1fe90e553571cf17f1a04e52afa6172b1a8